File: agent-app-go/internal/sql/schema_linker_test.go
package sql

import "testing"

func TestSchemaLinker_LinksQuestionValuesFromDbSchemaValueHintsContract(t *testing.T) {
	linker := &SchemaLinker{}
	schema := `{
		"tables": {"departments": ["name (character varying)"]},
		"valueHints": {"departments.name": ["플랫폼팀", "기술지원팀"]}
	}`

	hints := linker.LinkEntities(schema, "기술지원팀 직원 목록")

	if len(hints) != 1 {
		t.Fatalf("got %+v", hints)
	}
	if hints[0].Suggestion != "departments.name = '기술지원팀'" {
		t.Fatalf("got %q", hints[0].Suggestion)
	}
}

File: agent-app-ktor/src/test/kotlin/com/riwonace/agentktor/sql/SchemaLinkerTest.kt
package com.riwonace.agentktor.sql

import com.fasterxml.jackson.databind.ObjectMapper
import kotlin.test.Test
import kotlin.test.assertEquals

class SchemaLinkerTest {

    private val linker = SchemaLinker(ObjectMapper())

    @Test
    fun `db schema Resource의 valueHints 계약에서 질문의 값을 연결한다`() {
        val schema = """
            {
              "tables": {"departments": ["name (character varying)"]},
              "valueHints": {"departments.name": ["플랫폼팀", "기술지원팀"]}
            }
        """.trimIndent()

        val hints = linker.linkEntities(schema, "기술지원팀 직원 목록")

        assertEquals(1, hints.size)
        assertEquals("departments.name = '기술지원팀'", hints.single().suggestion)
    }
}
